src/content/docs/guides/Web development.mdx

```diff
@@ -148,12 +150,81 @@ Tutaj mamy dwie ścieżki:
 
 1. React Server Components
 
-   Domyślna ścieżka, nie trzeba zarządzać stanami ładowania, wszystko jest od razu na miejscu, bardzo wygodne
+   React Server Components (RSC) to domyślne podejście w Next.js 13+. Dzięki nim dane są pobierane po stronie serwera, co eliminuje konieczność zarządzania stanami ładowania i zapewnia szybsze ładowanie stron. Zaletami jest automatyczne renderowanie danych na serwerze, brak konieczności zarządzania stanami ładowania i są proste w użyciu.
+
+    ```jsx
+    // app/products/page.tsx
+    import { db } from '@/lib/db';
+
+    export default async function ProductsPage() {
+      // Pobieranie danych z bazy po stronie serwera
+      const products = await db.products.findMany();
+
+      return (
+        <div>
+          <h1>Produkty</h1>
+          <ul>
+            {products.map((product) => (
+              <li key={product.id}>
+                {product.name} - {product.price} PLN
+              </li>
+            ))}
+          </ul>
+        </div>
+      );
+    }
+    ```
 
 2. Tanstack Query
 
    Tego używamy gdy chcemy mieć większą kontrolę nad tym co się dzieje w naszej aplikacji, potrzebujemy odświeżać często dane albo na przykład mamy jakąś skomplikowaną logikę po stronie klienta (typu użytkownik najpierw wybiera jakąś zakładkę i dopiero wtedy pobieramy coś z api albo jakikolwiek case z bardziej zaawansowanym filtrowaniem)
 
+    ```ts {2, 11}
+    // actions/products.ts
+    "use server"
+
+    import { db } from '@/lib/db';
+
+    /** 
+     * PAMIĘTAJCIE, że tego typu akcje serwerowe są automatycznie generowanymi
+     * endpointami API, więc każdy może przy odrobinie umiejętności wykonać swoje rządanie do tej funkcji.
+     * Trzeba zawsze zabezpieczyć taką funkcję, jeśli robimy rzeczy tylko dla zalogowanych użytkowników, np zmiana nazwy.
+    */
+    export async function fetchProducts() {
+      return await db.products.findMany(); // Przykład z bazą danych
+    }
+    ```
+
+    ```jsx {2, 10}
+    // app/products/page.tsx
+    "use client"
+
+    import { useQuery } from '@tanstack/react-query';
+    import { fetchProducts } from '@/actions/products.ts';
+
+    export default function ProductsList() {
+      const { data: products, isLoading, error, refetch } = useQuery({
+        queryKey: ['products'],
+        queryFn: fetchProducts,
+        refetchOnMount: false, // zalezy od twoich potrzeb
```

Comment:
albo bym to usunął (raczej to), albo rozpisał dokładniej o co z tym chodzi, ja pierwszy raz się spotykam z tym szczerze

Reply:
ewentualnie jak jesteś ambitny to możesz zrobić przykład z prostą paginacją w useState, bo to chyba najlepszy usecase do uzywania useQuery

Reply:
why usunac?

Reply:
w sensie to `refetchOnMount`, idk dlaczego akurat to dałeś 😭